addons/survey/models/survey_survey.py

```diff
@@ -189,14 +190,33 @@ def _compute_survey_statistic(self):
             if item['scoring_success']:
                 stat[item['survey_id'][0]]['success_count'] += item['__count']
 
-        for survey_id, values in stat.items():
-            avg_total = stat[survey_id].pop('answer_score_avg_total')
-            stat[survey_id]['answer_score_avg'] = avg_total / (stat[survey_id]['answer_done_count'] or 1)
-            stat[survey_id]['success_ratio'] = (stat[survey_id]['success_count'] / (stat[survey_id]['answer_done_count'] or 1.0))*100
+        for survey_stats in stat.values():
+            avg_total = survey_stats.pop('answer_score_avg_total')
+            survey_stats['answer_score_avg'] = avg_total / (survey_stats['answer_done_count'] or 1)
+            survey_stats['success_ratio'] = (survey_stats['success_count'] / (survey_stats['answer_done_count'] or 1.0))*100
 
         for survey in self:
             survey.update(stat.get(survey._origin.id, default_vals))
 
+    @api.depends('user_input_ids.survey_id', 'user_input_ids.start_datetime', 'user_input_ids.end_datetime')
+    def _compute_answer_duration_avg(self):
+        result_per_survey_id = {}
+        if self.ids:
+            self.env.cr.execute(
+                """SELECT survey_id,
+                          avg((extract(epoch FROM end_datetime)) - (extract (epoch FROM start_datetime)))
```

Comment:
What happen if `end_datetime` is null ?
I think in that case the entire average will be null, should we add a condition to use only records with not null end/start datetime ?
(In practice I do not think it's possible to have survey_user_input "done" without end_datetime, but I would add the condition just in case)